src/iarc7_motion/iarc_tasks/position_holder.py

```diff
@@ -0,0 +1,109 @@
+        #self._update_period = 1.0/update_rate
+        self._update_period = 0.1
+        self._lock = threading.RLock()
+        self._odometry = None
+        self._hold_x = x
+        self._hold_y = y
+        self._max_acceleration = rospy.get_param('~max_translation_acceleration', 0.0)
+        self._max_speed = rospy.get_param('~max_translation_speed', 0.0)
+        self._position_tolerance = rospy.get_param('~translation_position_hold_tolerance', 0.0)
+        self._current_velocity_sub = rospy.Subscriber('/odometry/filtered',
+                                              Odometry,
+                                              self._current_velocity_callback)
+
+    def get_xy_hold_response(self, current_x, current_y, z_velocity=None):
+        with self._lock:
+            if self._odometry is not None:
+                delta_x = current_x - self._hold_x
+                delta_y = current_y - self._hold_y
+                response = self._get_next_acceleration_target_trapezoidal(delta_x,
+                                                                          delta_y,
+                                                                          self._odometry.twist,
+                                                                          z_velocity=z_velocity)
+            else:
+                rospy.logerr('get_xy_hold_response called before odometry published')
+
+            return response
+
+    def _current_velocity_callback(self, odometry):
+        with self._lock:
+            if odometry.header.frame_id == 'map' and odometry.child_frame_id == 'level_quad':
+                self._odometry = odometry
+                if self._hold_x is None:
+                    self._hold_x = odometry.pose.pose.position.x
+                if self._hold_y is None:
+                    self._hold_y = odometry.pose.pose.position.y
+
+    def _calculate_trapezoidal_acceleration(self, distance, speed):
+        # Check if we can decelerate in time
+        try:
+            target_acceleration = (speed**2)/(2*distance)
+        except ZeroDivisionError:
+            # We are at the target point, no need to accelerate anywhere
+            return 0.0
+
+        if target_acceleration < self._max_acceleration:
+            if speed < self._max_speed:
+                target_acceleration = self._max_acceleration
+            else:
+                target_acceleration = 0
+        else:
+            target_acceleration = -self._max_acceleration
+
+        return target_acceleration
+
+    def _get_next_acceleration_target_trapezoidal(self, x, y, current_twist, z_velocity=None):
+        # calculate current straight line velocity
+        speed = math.sqrt(current_twist.twist.linear.x**2 + current_twist.twist.linear.y**2)
+        # calculate straight line distance
+        distance = math.sqrt(x**2 + y**2)
+
+        target_acceleration = self._calculate_trapezoidal_acceleration(distance, speed)
```

Comment:
I think that we shouldn't be worried about trapezoidal acceleration in the direction perpendicular to our direction of movement, we should request whatever acceleration we need to stay on track in that direction.  This is tied to the larger discussion about path planning in general.